src/connection_string.ts

```diff
@@ -255,6 +255,15 @@ export function parseOptions(
     mongoClient = undefined;
   }
 
+  // validate BSONOptions
+  if (options.useBigInt64 && !options.promoteLongs) {
+    throw new MongoAPIError('Must request either bigint or Long for int64 deserialization');
+  }
+
+  if (options.useBigInt64 && !options.promoteValues) {
+    throw new MongoAPIError('Must request either bigint or Long for int64 deserialization');
+  }
```

Comment:
unsure, the error messages are different.  but we'll never hit the second if-statement anyways.  Also I'm not sure this is the right place for these errors. Both of these options can be set per operation, so if we want to prevent users from using both options concurrently we need to put the logic in a place where every operation will encounter it.

also - it might be worthwhile to check for "existence" instead of truthiness here for `promoteValues`

Reply:
No, one is comparing against `options.promoteLongs` and the other is against `options.promoteValues`

Reply:
follow-up - where do we test that combining options throws errors from the bson layer when the options are set for DB, collection, or operation? We only have tests for these two particular errors for the mongo client

Reply:
Good point here. In actuality though, throwing here duplicates logic from the bson library itself. If either one of these combinations of options were used in a call to BSON.deserialize, we would get a BSONError with the same message.

That makes me think removing this logic altogether is likely the best approach as having this logic here does imply we should have it in every other place you mentioned, but they are [already covered](https://github.com/mongodb/js-bson/blob/b6a15b5a5eeb0cf453f85fb5bbfeaf3add110015/src/parser/deserializer.ts#L127-L133).

@durran What are your thoughts?

Reply:
I think it's worthwhile to leave this logic here.

In this case, if both options are provided, errors will be thrown when the client is constructed (before any operations are run) if we remove this logic, no errors are surfaced for invalid client options until the first command is executed.

But we should still document with tests that we expect the bson library to throw these errors when executing a command with both of these arguments provided.